### PR TITLE
Add authorization to ConversationController, CustomerController, MessageController

### PR DESCRIPTION
**Pull request by wmnnd**
Adds plug-based authorization for show/update/delete routes as suggested
in #638.
Also as suggested in #638, uses 404 status code when no authorized
resource was found.

### Issue
#638 

## Checklist

- [X] Everything passes when running `mix test`
- [X] Ran `mix format`

### CODE REVIEW

**Comment by wmnnd** [comment]
The scope of this PR is currently as described in #638, i.e. only implementing authorization for the `:show`/`:update`/`:delete` routes. Authorization should probably be added to all other routes of the affected controllers as well.

File: lib/chat_api_web/controllers/customer_controller.ex [hunk 1]
```diff
@@ -8,6 +8,25 @@ defmodule ChatApiWeb.CustomerController do
 
   action_fallback(ChatApiWeb.FallbackController)
 
+  plug :authorize when action in [:show, :update, :delete]
+
+  defp authorize(conn, _) do
+    id = conn.path_params["id"]
+
+    preloads =
+      (conn.path_params["expand"] || ["company", "tags"])
```
Comment (reichert621) [inline lib/chat_api_web/controllers/customer_controller.ex:17]: hmm i wonder if we can remove the "company"/"tags" defaults... probably safer to keep them in for now 👍 (just commenting so i don't forget)

**Review by reichert621 — APPROVED** [review]
thanks @wmnnd! cleans things up nicely 🎉 🎉 🎉

File: lib/chat_api_web/controllers/customer_controller.ex (hunk 1 above)
Comment (wmnnd) [reply lib/chat_api_web/controllers/customer_controller.ex:17]: It’s from the old `get_customer!/1` and removing this default seems to break three or four tests.

Comment (reichert621) [reply lib/chat_api_web/controllers/customer_controller.ex:17]: makes sense!

**Pull request merged**